src/Microsoft.Diagnostics.Monitoring.WebApi/Operation/EgressOperationService.cs

```diff
@@ -46,7 +46,12 @@ private async Task ExecuteEgressOperation(EgressRequest egressRequest, Cancellat
 
                 try
                 {
-                    ExecutionResult<EgressResult> result = await egressRequest.EgressOperation.ExecuteAsync(_serviceProvider, token);
+                    Task<ExecutionResult<EgressResult>> executeTask = egressRequest.EgressOperation.ExecuteAsync(_serviceProvider, token);
+
+                    await egressRequest.EgressOperation.Started.WithCancellation(token);
```

Comment:
Why can't this and the marking the operation as running be part of `EgressOperation.ExecuteAsync`? This currently feels like it's leaking the details of the operation execution outside of where it needs to be understood.

Reply:
Nevermind, sorry, I misunderstood that the execution state is actually stored within the operation store.